lib/matplotlib/pyplot.py

```diff
@@ -1108,15 +1184,15 @@ def delaxes(ax=None):
     ax.remove()
 
 
-def sca(ax):
+def sca(ax: Axes) -> None:
     """
     Set the current Axes to *ax* and the current Figure to the parent of *ax*.
     """
     figure(ax.figure)
-    ax.figure.sca(ax)
+    ax.figure.sca(ax)  # type: ignore
```

Comment:
I want to say its the minor difference between `_AxesBase` and `Axes`, but I'll make sure and either remove the type ignore or add a more helpful comment

Reply:
It was actually that `ax.figure` is typed as `FigureCanvasBase | None`, and None (of course) does not have an `sca` method.

But if you are calling `plt.sca` with an axes not in a figure already, that is its own problem.

The alternative would be to raise `if ax.figure is None:` which would I think quiet mypy because of type narrowing.

Reply:
added the aforementioned comment about what was being ignored in d8329509